### ISSUE TITLE
crushed when training the network

### ISSUE THREAD

**Issue by JackWang1997** [tawnkramer/sdsandbox#22]
Hi I changed to the donkey car and then i felt that maybe i could just modify the sdsandbox a bit to get what i want. I did a few changes in the past few weeks and then i decided to start everything again. (guess i messed up so many codes and i forgot how to change back) So i download everything again. However this time i found that it crushed during the training process. It gave back things like this:

working on model outputs/3333.h5
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
cropping2d (Cropping2D)      (None, 80, 160, 3)        0         
_________________________________________________________________
batch_normalization (BatchNo (None, 80, 160, 3)        12        
_________________________________________________________________
conv2d (Conv2D)              (None, 38, 78, 24)        1824      
_________________________________________________________________
dropout (Dropout)            (None, 38, 78, 24)        0         
_________________________________________________________________
conv2d_1 (Conv2D)            (None, 17, 37, 32)        19232     
_________________________________________________________________
dropout_1 (Dropout)          (None, 17, 37, 32)        0         
_________________________________________________________________
conv2d_2 (Conv2D)            (None, 7, 17, 48)         38448     
_________________________________________________________________
dropout_2 (Dropout)          (None, 7, 17, 48)         0         
_________________________________________________________________
conv2d_3 (Conv2D)            (None, 3, 8, 64)          27712     
_________________________________________________________________
dropout_3 (Dropout)          (None, 3, 8, 64)          0         
_________________________________________________________________
conv2d_4 (Conv2D)            (None, 1, 6, 64)          36928     
_________________________________________________________________
dropout_4 (Dropout)          (None, 1, 6, 64)          0         
_________________________________________________________________
flatten (Flatten)            (None, 384)               0         
_________________________________________________________________
dense (Dense)                (None, 100)               38500     
_________________________________________________________________
dropout_5 (Dropout)          (None, 100)               0         
_________________________________________________________________
dense_1 (Dense)              (None, 50)                5050      
_________________________________________________________________
dropout_6 (Dropout)          (None, 50)                0         
_________________________________________________________________
dense_2 (Dense)              (None, 2)                 102       
=================================================================
Total params: 167,808
Trainable params: 167,802
Non-trainable params: 6
_________________________________________________________________
(None, 80, 160, 3)
(None, 80, 160, 3)
(None, 38, 78, 24)
(None, 38, 78, 24)
(None, 17, 37, 32)
(None, 17, 37, 32)
(None, 7, 17, 48)
(None, 7, 17, 48)
(None, 3, 8, 64)
(None, 3, 8, 64)
(None, 1, 6, 64)
(None, 1, 6, 64)
(None, 384)
(None, 100)
(None, 100)
(None, 50)
(None, 50)
(None, 2)
found 3059 files
num train/val 2427 632
steps_per_epoch 37 validation_steps 9
Epoch 1/100
2019-04-27 20:54:35.844897: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2
2019-04-27 20:54:35.845623: I tensorflow/core/common_runtime/process_util.cc:69] Creating new thread pool with default inter op setting: 16. Tune using inter_op_parallelism_threads for best performance.
37/37 [==============================] - 57s 2s/step - loss: 0.0101 - val_loss: 0.0155
Epoch 2/100
37/37 [==============================] - 50s 1s/step - loss: 0.0067 - val_loss: 0.0137
Epoch 3/100
37/37 [==============================] - 50s 1s/step - loss: 0.0052 - val_loss: 0.0072
Epoch 4/100
37/37 [==============================] - 50s 1s/step - loss: 0.0046 - val_loss: 0.0052
Epoch 5/100
37/37 [==============================] - 49s 1s/step - loss: 0.0047 - val_loss: 0.0037
Epoch 6/100
37/37 [==============================] - 50s 1s/step - loss: 0.0041 - val_loss: 0.0034
Epoch 7/100
37/37 [==============================] - 49s 1s/step - loss: 0.0042 - val_loss: 0.0033
Epoch 8/100
37/37 [==============================] - 50s 1s/step - loss: 0.0041 - val_loss: 0.0034
Epoch 9/100
37/37 [==============================] - 49s 1s/step - loss: 0.0040 - val_loss: 0.0035
Epoch 10/100
37/37 [==============================] - 50s 1s/step - loss: 0.0039 - val_loss: 0.0031
Epoch 11/100
37/37 [==============================] - 49s 1s/step - loss: 0.0037 - val_loss: 0.0032
Epoch 12/100
37/37 [==============================] - 42s 1s/step - loss: 0.0040 - val_loss: 0.0030
Epoch 13/100
37/37 [==============================] - 40s 1s/step - loss: 0.0037 - val_loss: 0.0029
Epoch 14/100
37/37 [==============================] - 40s 1s/step - loss: 0.0034 - val_loss: 0.0028
Epoch 15/100
37/37 [==============================] - 40s 1s/step - loss: 0.0036 - val_loss: 0.0025
Epoch 16/100
37/37 [==============================] - 40s 1s/step - loss: 0.0035 - val_loss: 0.0023
Epoch 17/100
37/37 [==============================] - 40s 1s/step - loss: 0.0033 - val_loss: 0.0028
Epoch 18/100
37/37 [==============================] - 40s 1s/step - loss: 0.0036 - val_loss: 0.0029
Epoch 19/100
37/37 [==============================] - 40s 1s/step - loss: 0.0034 - val_loss: 0.0029
Epoch 20/100
37/37 [==============================] - 40s 1s/step - loss: 0.0033 - val_loss: 0.0028
Epoch 21/100
37/37 [==============================] - 40s 1s/step - loss: 0.0032 - val_loss: 0.0026
Epoch 22/100
37/37 [==============================] - 40s 1s/step - loss: 0.0033 - val_loss: 0.0027
Epoch 23/100
37/37 [==============================] - 40s 1s/step - loss: 0.0031 - val_loss: 0.0027
Epoch 24/100
37/37 [==============================] - 40s 1s/step - loss: 0.0030 - val_loss: 0.0024
Epoch 25/100
37/37 [==============================] - 40s 1s/step - loss: 0.0034 - val_loss: 0.0024
Epoch 26/100
37/37 [==============================] - 40s 1s/step - loss: 0.0029 - val_loss: 0.0027
2019-04-27 21:13:49.730 python[48507:2609141] -[NSApplication _setup:]: unrecognized selector sent to instance 0x7fcd41211820
2019-04-27 21:13:49.734 python[48507:2609141] An uncaught exception was raised
2019-04-27 21:13:49.734 python[48507:2609141] -[NSApplication _setup:]: unrecognized selector sent to instance 0x7fcd41211820
2019-04-27 21:13:49.734 python[48507:2609141] (
	0   CoreFoundation                      0x00007fff92dae4f2 __exceptionPreprocess + 178
	1   libobjc.A.dylib                     0x00007fff91de673c objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff92e181ad -[NSObject(NSObject) doesNotRecognizeSelector:] + 205
	3   CoreFoundation                      0x00007fff92d1e571 ___forwarding___ + 1009
	4   CoreFoundation                      0x00007fff92d1e0f8 _CF_forwarding_prep_0 + 120
	5   libtk8.6.dylib                      0x0000000b3eda831d TkpInit + 413
	6   libtk8.6.dylib                      0x0000000b3ed0017e Initialize + 2622
	7   _tkinter.cpython-36m-darwin.so      0x0000000b3abeea16 _tkinter_create + 1174
	8   python                              0x000000010e0d6068 _PyCFunction_FastCallDict + 200
	9   python                              0x000000010e1ab61f call_function + 143
	10  python                              0x000000010e1a9175 _PyEval_EvalFrameDefault + 46837
	11  python                              0x000000010e19c8c9 _PyEval_EvalCodeWithName + 425
	12  python                              0x000000010e1ac2cc _PyFunction_FastCallDict + 364
	13  python                              0x000000010e054f80 _PyObject_FastCallDict + 320
	14  python                              0x000000010e07c5f8 method_call + 136
	15  python                              0x000000010e05c5ce PyObject_Call + 62
	16  python                              0x000000010e0fd5b5 slot_tp_init + 117
	17  python                              0x000000010e101af1 type_call + 241
	18  python                              0x000000010e054ef1 _PyObject_FastCallDict + 177
	19  python                              0x000000010e05d137 _PyObject_FastCallKeywords + 327
	20  python                              0x000000010e1ab718 call_function + 392
	21  python                              0x000000010e1a9225 _PyEval_EvalFrameDefault + 47013
	22  python                              0x000000010e1ab9dc fast_function + 188
	23  python                              0x000000010e1ab67c call_function + 236
	24  python                              0x000000010e1a9175 _PyEval_EvalFrameDefault + 46837
	25  python                              0x000000010e19c8c9 _PyEval_EvalCodeWithName + 425
	26  python                              0x000000010e1ac2cc _PyFunction_FastCallDict + 364
	27  python                              0x000000010e054f80 _PyObject_FastCallDict + 320
	28  python                              0x000000010e07c5f8 method_call + 136
	29  python                              0x000000010e05c5ce PyObject_Call + 62
	30  python                              0x000000010e1a9376 _PyEval_EvalFrameDefault + 47350
	31  python                              0x000000010e19c8c9 _PyEval_EvalCodeWithName + 425
	32  python                              0x000000010e1aba8a fast_function + 362
	33  python                              0x000000010e1ab67c call_function + 236
	34  python                              0x000000010e1a9175 _PyEval_EvalFrameDefault + 46837
	35  python                              0x000000010e1ab9dc fast_function + 188
	36  python                              0x000000010e1ab67c call_function + 236
	37  python                              0x000000010e1a9175 _PyEval_EvalFrameDefault + 46837
	38  python                              0x000000010e19c8c9 _PyEval_EvalCodeWithName + 425
	39  python                              0x000000010e1aba8a fast_function + 362
	40  python                              0x000000010e1ab67c call_function + 236
	41  python                              0x000000010e1a9175 _PyEval_EvalFrameDefault + 46837
	42  python                              0x000000010e19c8c9 _PyEval_EvalCodeWithName + 425
	43  python                              0x000000010e1aba8a fast_function + 362
	44  python                              0x000000010e1ab67c call_function + 236
	45  python                              0x000000010e1a9175 _PyEval_EvalFrameDefault + 46837
	46  python                              0x000000010e19c8c9 _PyEval_EvalCodeWithName + 425
	47  python                              0x000000010e1aba8a fast_function + 362
	48  python                              0x000000010e1ab67c call_function + 236
	49  python                              0x000000010e1a9225 _PyEval_EvalFrameDefault + 47013
	50  python                              0x000000010e19c8c9 _PyEval_EvalCodeWithName + 425
	51  python                              0x000000010e1f555c PyRun_FileExFlags + 252
	52  python                              0x000000010e1f4a34 PyRun_SimpleFileExFlags + 372
	53  python                              0x000000010e21b7c6 Py_Main + 3734
	54  python                              0x000000010e04cf59 main + 313
	55  libdyld.dylib                       0x00007fff8f1785ad start + 1
	56  ???                                 0x0000000000000003 0x0 + 3
)
2019-04-27 21:13:49.735 python[48507:2609141] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[NSApplication _setup:]: unrecognized selector sent to instance 0x7fcd41211820'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff92dae4f2 __exceptionPreprocess + 178
	1   libobjc.A.dylib                     0x00007fff91de673c objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff92e181ad -[NSObject(NSObject) doesNotRecognizeSelector:] + 205
	3   CoreFoundation                      0x00007fff92d1e571 ___forwarding___ + 1009
	4   CoreFoundation                      0x00007fff92d1e0f8 _CF_forwarding_prep_0 + 120
	5   libtk8.6.dylib                      0x0000000b3eda831d TkpInit + 413
	6   libtk8.6.dylib                      0x0000000b3ed0017e Initialize + 2622
	7   _tkinter.cpython-36m-darwin.so      0x0000000b3abeea16 _tkinter_create + 1174
	8   python                              0x000000010e0d6068 _PyCFunction_FastCallDict + 200
	9   python                              0x000000010e1ab61f call_function + 143
	10  python                              0x000000010e1a9175 _PyEval_EvalFrameDefault + 46837
	11  python                              0x000000010e19c8c9 _PyEval_EvalCodeWithName + 425
	12  python                              0x000000010e1ac2cc _PyFunction_FastCallDict + 364
	13  python                              0x000000010e054f80 _PyObject_FastCallDict + 320
	14  python                              0x000000010e07c5f8 method_call + 136
	15  python                              0x000000010e05c5ce PyObject_Call + 62
	16  python                              0x000000010e0fd5b5 slot_tp_init + 117
	17  python                              0x000000010e101af1 type_call + 241
	18  python                              0x000000010e054ef1 _PyObject_FastCallDict + 177
	19  python                              0x000000010e05d137 _PyObject_FastCallKeywords + 327
	20  python                              0x000000010e1ab718 call_function + 392
	21  python                              0x000000010e1a9225 _PyEval_EvalFrameDefault + 47013
	22  python                              0x000000010e1ab9dc fast_function + 188
	23  python                              0x000000010e1ab67c call_function + 236
	24  python                              0x000000010e1a9175 _PyEval_EvalFrameDefault + 46837
	25  python                              0x000000010e19c8c9 _PyEval_EvalCodeWithName + 425
	26  python                              0x000000010e1ac2cc _PyFunction_FastCallDict + 364
	27  python                              0x000000010e054f80 _PyObject_FastCallDict + 320
	28  python                              0x000000010e07c5f8 method_call + 136
	29  python                              0x000000010e05c5ce PyObject_Call + 62
	30  python                              0x000000010e1a9376 _PyEval_EvalFrameDefault + 47350
	31  python                              0x000000010e19c8c9 _PyEval_EvalCodeWithName + 425
	32  python                              0x000000010e1aba8a fast_function + 362
	33  python                              0x000000010e1ab67c call_function + 236
	34  python                              0x000000010e1a9175 _PyEval_EvalFrameDefault + 46837
	35  python                              0x000000010e1ab9dc fast_function + 188
	36  python                              0x000000010e1ab67c call_function + 236
	37  python                              0x000000010e1a9175 _PyEval_EvalFrameDefault + 46837
	38  python                              0x000000010e19c8c9 _PyEval_EvalCodeWithName + 425
	39  python                              0x000000010e1aba8a fast_function + 362
	40  python                              0x000000010e1ab67c call_function + 236
	41  python                              0x000000010e1a9175 _PyEval_EvalFrameDefault + 46837
	42  python                              0x000000010e19c8c9 _PyEval_EvalCodeWithName + 425
	43  python                              0x000000010e1aba8a fast_function + 362
	44  python                              0x000000010e1ab67c call_function + 236
	45  python                              0x000000010e1a9175 _PyEval_EvalFrameDefault + 46837
	46  python                              0x000000010e19c8c9 _PyEval_EvalCodeWithName + 425
	47  python                              0x000000010e1aba8a fast_function + 362
	48  python                              0x000000010e1ab67c call_function + 236
	49  python                              0x000000010e1a9225 _PyEval_EvalFrameDefault + 47013
	50  python                              0x000000010e19c8c9 _PyEval_EvalCodeWithName + 425
	51  python                              0x000000010e1f555c PyRun_FileExFlags + 252
	52  python                              0x000000010e1f4a34 PyRun_SimpleFileExFlags + 372
	53  python                              0x000000010e21b7c6 Py_Main + 3734
	54  python                              0x000000010e04cf59 main + 313
	55  libdyld.dylib                       0x00007fff8f1785ad start + 1
	56  ???                                 0x0000000000000003 0x0 + 3
)
libc++abi.dylib: terminating with uncaught exception of type NSException
Abort trap: 6

As you can see it happened during the training process. Is it because i installed something wrong since i switched to a new computer?

**Comment by tawnkramer** [comment]
I don't use TF on a Mac. You will need to ask TF people.
